### PR TITLE
Proxy content as binary content to support images and other octets

### PR DESCRIPTION
**Pull request by dluc**
# Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:
- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

# Description of the change
Change the implementation to proxy the content in a more opaque way, in order to support binary transfers, e.g. images.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/reverse-proxy-dotnet/19)

### CODE REVIEW

File: ProxyAgent.Test/ProxyTest.cs [hunk 1]
```diff
@@ -128,30 +130,34 @@ public void ItDoesntProxySomeHeaders()
...
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("X-ActivityId"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => r.Headers.Contains("X-CorrelationId"))), Times.Once);
+
+            // Assert - these request headers are blocked
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Connection"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Content-Length"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Keep-Alive"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Host"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Upgrade"))), Times.Once);
             this.client.Verify(x => x.GetAsync(It.Is<IHttpRequest>(r => !r.Headers.Contains("Upgrade-Insecure-Requests"))), Times.Once);
```
Comment (timlaverty) [inline ProxyAgent.Test/ProxyTest.cs]: #supernit Unsecure, not Insecure

File: ProxyAgent/HttpClient/HttpClient.cs [hunk 2]
```diff
@@ -124,7 +125,7 @@ public HttpClient(ILogger logger, IConfig config)
                         {
                             StatusCode = response.StatusCode,
                             Headers = headers,
-                            Content = await response.Content.ReadAsStringAsync()
+                            Content = await response.Content.ReadAsByteArrayAsync()
```
Comment (timlaverty) [inline ProxyAgent/HttpClient/HttpClient.cs]: so everything will go back as byte array.

Comment (dluc) [reply ProxyAgent/HttpClient/HttpClient.cs]: 👍

File: ProxyAgent.Test/ProxyTest.cs (hunk 1 above)
Comment (dluc) [reply ProxyAgent.Test/ProxyTest.cs]: can't fix the web :-) https://www.w3.org/TR/upgrade-insecure-requests

**Pull request merged**